### ISSUE TITLE
untyped licenses

### ISSUE THREAD

**Issue by MasseGuillaume** [cvogt/cbt#67]
currently licences accepts any (String, Url)
https://github.com/cvogt/cbt/blob/6d5095e217a8c6e989a9df7cb62a6b2d5a857297/stage2/Lib.scala#L20

this causes a complete mess in practice:
https://github.com/metadoc/metadoc/tree/develop/observations/licences

I would suggest something like:
```scala
trait License { // << not sealed
  def url: Url
  def toString: String
}
Apache2
MIT
AGPL
GPL2
GPL3
LGPL
MPL2
BSD
// ...
```

Nix uses this approach: https://github.com/NixOS/nixpkgs/blob/master/lib/licenses.nix#L1 ([spdx standard](https://spdx.org/))

**Comment by cvogt** [comment]
Probably not the most pressing issue. But I'd happily merge a PR. How about
```scala
sealed trait License { // << not sealed
  def name: String
  def url: Url
}
final case class CustomLicense( name: String, url: String ) extends License
object Apache2 extends CustomLicense
...
```

How about putting it in `stage2/pom.scala` and moving `Developer` there as well from `Lib.scala`.

**Comment by MasseGuillaume** [comment]
Make sense, let's handle the most popular license. I will create a PR soon.

**Comment by cvogt** [comment]
make that sealed abstract class :)

**Comment by cvogt** [comment]
fixed in #143
